providers/Flipt/src/FliptProvider.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+namespace OpenFeature\Providers\Flipt;
+
+use Flipt\Client\FliptClient;
+use Flipt\Models\BooleanEvaluationResult;
+use Flipt\Models\VariantEvaluationResult;
+use OpenFeature\implementation\provider\AbstractProvider;
+use OpenFeature\implementation\provider\ResolutionDetailsBuilder;
+use OpenFeature\implementation\provider\ResolutionDetailsFactory;
+use OpenFeature\implementation\provider\ResolutionError;
+use OpenFeature\interfaces\flags\EvaluationContext;
+use OpenFeature\interfaces\flags\FlagValueType;
+use OpenFeature\interfaces\provider\ErrorCode;
+use OpenFeature\interfaces\provider\Provider;
+use OpenFeature\interfaces\provider\ResolutionDetails;
+use Psr\SimpleCache\CacheInterface;
+
+class FliptProvider extends AbstractProvider implements Provider
+{
+    protected const NAME = 'FliptProvider';
+
+    protected $client;
+    protected Cache $cache;
+
+    public function __construct( mixed $hostOrClient, string $apiToken = '', string $namespace = '', CacheInterface $cache = null ) {
```

Comment:
The minimum supported version of OpenFeature SDK is PHP 8.0, so you can used union types here.


```suggestion
    public function __construct(string|FliptClient $hostOrClient, string $apiToken = '', string $namespace = '', CacheInterface $cache = null ) {
```